src/d/actor/d_a_obj_doguu_demo.cpp

```diff
@@ -4,50 +4,145 @@
 //
 
 #include "d/actor/d_a_obj_doguu_demo.h"
+#include "d/d_bg_w.h"
+#include "d/d_cc_d.h"
+#include "d/d_com_inf_game.h"
+#include "d/d_procname.h"
 #include "dolphin/types.h"
+#include "f_op/f_op_actor.h"
+#include "f_op/f_op_actor_mng.h"
+#include "m_Do/m_Do_mtx.h"
 
-/* 00000078-00000098       .text CheckCreateHeap__FP10fopAc_ac_c */
-static BOOL CheckCreateHeap(fopAc_ac_c*) {
-    /* Nonmatching */
+static BOOL CheckCreateHeap(fopAc_ac_c* i_this) {
+    return ((daObjDoguuD_c*)i_this)->CreateHeap();
 }
 
-/* 00000098-00000214       .text CreateHeap__13daObjDoguuD_cFv */
-void daObjDoguuD_c::CreateHeap() {
-    /* Nonmatching */
+BOOL daObjDoguuD_c::CreateHeap() {
+    this->param = (u8)mBase.mParameters;  // Im not sure why this is here, none of the other classes
```

Comment:
```suggestion
    mUnusedParam = fopAcM_GetParam(this) & 0xFF;
```
the parameters field is almost always accessed via fopAcM_GetParam, and if you see what resembles a cast to u8 when parameters are involved, it's probably actually masking out an 8-bit parameter manually like this.

member variables should ideally start with `m` to distinguish them from local variables.
you don't need to type `this->` when assigning to a member variable.

also, as far as I can see this & 0xFF parameter appears unused (correct me if I'm wrong), so it can have unused in its name to make that clearer